conftest.py

```diff
@@ -243,37 +242,35 @@ def selenium_standalone(request, web_server_main):
         try:
             yield selenium
         finally:
-            print(selenium.logs)
             selenium.driver.quit()
 
-    @pytest.fixture(params=["firefox", "chrome"], scope="module")
-    def _selenium_cached(request, web_server_main):
-        # Cached selenium instance. This is a copy-paste of
-        # selenium_standalone to avoid fixture scope issues
-        server_hostname, server_port, server_log = web_server_main
-        if request.param == "firefox":
-            cls = FirefoxWrapper
-        elif request.param == "chrome":
-            cls = ChromeWrapper
-        selenium = cls(
-            build_dir=request.config.option.build_dir,
-            server_port=server_port,
-            server_hostname=server_hostname,
-            server_log=server_log,
-        )
+    @pytest.fixture(params=["firefox", "chrome"], scope="function")
+    def selenium_standalone(request, web_server_main):
         try:
+            selenium = next(selenium_common(request, web_server_main))
```

Comment:
@rth Is this a correct way to compose generators? I'm not sure that the finally block code from `selenium_common` ever gets run...

Reply:
I think it doesn't.

Reply:
Something like,
```py
with selenium_common(..) as selenium:
    yield selenium
```
doesn't work?

Reply:
Okay, it does. Probably what happens in the generator destructor looks something like:
```py
if self.gi_running:
    self.throw(StopIteration(None))
```

Reply:
```py
with selenium_common(..) as selenium: 
   yield selenium
```
would work, but then I have to decorate `selenium_common` with `@contextlib.contextmanager`. Maybe that's the best approach.

Reply:
>  but only if we decorate selenium_common with @contextlib.contextmanager

+1

Reply:
Yeah okay what I said above about generator destructors is correct except instead of `StopIteration` it throws `GeneratorExit`.

Reply:
Doing
```py
with selenium_common(..) as selenium: 
   yield selenium
```
runs finalizers from inside out (which is probably what you expect), doing what I did runs the finalizers from outside in.